### ISSUE TITLE
undefined method `new' for RbConfig:Module

### ISSUE THREAD

**Issue by itsmechlark** [resque/resque#1236]
`1-x-stable` references, but does not provide `Resque::Config` leading to confusing exceptions when using `Redis::config=`, 'Redis::config`, or`Redis::configure`.

```
Johns-MacBook-Pro:current itsmechlark$ RAILS_ENV=development bundle exec rake resque:work --trace
/Users/itsmechlark/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/runtime.rb:222: warning: Insecure world writable dir /Users/itsmechlark/.apache/maven/3.1.1/bin in PATH, mode 040777
/Users/itsmechlark/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler.rb:301: warning: Insecure world writable dir /Users/itsmechlark/.apache/maven/3.1.1/bin in PATH, mode 040777
** Invoke resque:work (first_time)
** Invoke resque:preload (first_time)
** Invoke resque:setup (first_time)
** Invoke environment (first_time)
** Execute environment
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/resque-1.25.2/lib/resque.rb:59:in `config': Use RbConfig instead of obsolete and deprecated Config.
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/resque-1.25.2/lib/resque.rb:59:in `config': Use RbConfig instead of obsolete and deprecated Config.
rake aborted!
NoMethodError: undefined method `new' for RbConfig:Module
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rbconfig/obsolete.rb:20:in `method_missing'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/resque-1.25.2/lib/resque.rb:59:in `config'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/resque-1.25.2/lib/resque.rb:63:in `configure'
/Users/itsmechlark/Desktop/GitHub/itsmechlark/cms/config/initializers/resque.rb:6:in `<top (required)>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:241:in `load'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:241:in `block in load'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:232:in `load_dependency'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:241:in `load'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/engine.rb:648:in `block in load_config_initializer'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/notifications.rb:161:in `instrument'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/engine.rb:647:in `load_config_initializer'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/engine.rb:611:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/engine.rb:611:in `block in <class:Engine>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:30:in `instance_exec'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:30:in `run'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:55:in `block in run_initializers'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:226:in `block in tsort_each'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:418:in `block (2 levels) in each_strongly_connected_component_from'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:427:in `each_strongly_connected_component_from'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:417:in `block in each_strongly_connected_component_from'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:44:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:44:in `tsort_each_child'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:411:in `call'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:411:in `each_strongly_connected_component_from'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:347:in `block in each_strongly_connected_component'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `call'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:345:in `each_strongly_connected_component'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:224:in `tsort_each'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/tsort.rb:205:in `tsort_each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/initializable.rb:54:in `run_initializers'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/application.rb:300:in `initialize!'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/railtie.rb:194:in `public_send'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/railtie.rb:194:in `method_missing'
/Users/itsmechlark/Desktop/GitHub/itsmechlark/cms/config/environment.rb:8:in `<top (required)>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:247:in `require'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:247:in `block in require'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:232:in `load_dependency'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/activesupport-4.1.4/lib/active_support/dependencies.rb:247:in `require'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/application.rb:276:in `require_environment!'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/railties-4.1.4/lib/rails/application.rb:379:in `block in run_tasks_blocks'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:240:in `call'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:240:in `block in execute'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:235:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:235:in `execute'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:201:in `block in invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:199:in `invoke_prerequisites'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:178:in `block in invoke_with_call_chain'
/Users/itsmechlark/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/task.rb:165:in `invoke'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:150:in `invoke_task'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:106:in `each'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:106:in `block in top_level'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:115:in `run_with_threads'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:100:in `top_level'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:78:in `block in run'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:176:in `standard_exception_handling'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/lib/rake/application.rb:75:in `run'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/gems/rake-10.3.2/bin/rake:33:in `<top (required)>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/bin/rake:23:in `load'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/bin/rake:23:in `<main>'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `eval'
/Users/itsmechlark/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `<main>'
```

<b>config/initializers/resque.rb</b>

```
require 'yaml'

config = YAML::load(File.open("#{Rails.root}/config/redis.yml"))[Rails.env]
Resque.redis = Redis.new(:host => config['host'], :port => config['port'])

Resque.configure do |config|
  config.pid_file = 'tmp/pids/resque.pid'
  config.interval = 1
  config.daemon = true
  config.queues = '*'
  config.graceful_term = true
  config.timeout = 5
end
```

ruby 2.1.2p95 (2014-05-08 revision 45877) [x86_64-darwin13.0]
ruby-2.1.0
ruby-1.9.3-p547
rails 4.1.4
resque 1.25.2

References:
[Resque configuration for Heroku Rails app](http://kennethjiang.blogspot.com/2014/02/resque-configuration-for-heroku-rails.html)
[Travis Build for ruby 2.1.2](https://travis-ci.org/itsmechlark/resque/jobs/30159455)

![alt text](https://travis-ci.org/itsmechlark/resque.svg?branch=master)
Thank you!

**Comment by dfockler** [comment]
Just ran into the same issue. I'm trying to update a plugin that uses `Resque.redis` which doesn't exist in `2.0.0`, but `Resque.config.redis` doesn't *technically* exist in `1-x-stable`. I'm just catching the error for now, but it dumps out a lot of RbConfig warnings

**Comment by jeremy** [comment]
Fixed by #1276
